include/tune_quda.h

```diff
@@ -67,7 +67,23 @@ namespace quda {
     virtual unsigned int minThreads() const { return 1; }
     virtual bool tuneGridDim() const { return true; }
     virtual bool tuneAuxDim() const { return false; }
-    virtual bool tuneSharedBytes() const { return true; }
+
+    virtual bool tuneSharedBytes() const
+    {
+      static bool tune_shared = true;
+      static bool init = false;
+
+      if (!init) {
+        char *enable_shared_env = getenv("QUDA_ENABLE_TUNING_SHARED");
```

Comment:
What was the motivation for adding this?

Regardless, I think it would better to query this environment variable once at startup (in `initQuda`, or wherever), then have this virtual function query that setting instead of having to call `getenv` each time execution hits a new tunable class.